services/justice_board.service.ts

```diff
@@ -0,0 +1,20 @@
+import { getDutiesByQuery } from '../repositories/duty.repository';
+import { getSoldiersByQuery } from '../repositories/soldier.repository';
+import { Soldier } from '../schemas/soldier.zschema';
+
+async function getJusticeBoard() {
```

Comment:
This whole function should be implemented using `projection` and `aggregation`

Reply:
```suggestion
const getJusticeBoard = getJusticeBoardFromRepository;
```

Reply:
Which you can even put in the export